packages/venia-concept/src/index.js

```diff
@@ -11,15 +11,18 @@ import setWithPath from 'lodash.set';
 import MutationQueueLink from '@adobe/apollo-link-mutation-queue';
 
 import { Util } from '@magento/peregrine';
-import { Adapter } from '@magento/venia-drivers';
+import { Adapter } from '@magento/venia-ui/lib/drivers';
 import store from './store';
 import app from '@magento/peregrine/lib/store/actions/app';
 import App, { AppContextProvider } from '@magento/venia-ui/lib/components/App';
+import StyleContextProvider from '@magento/peregrine/lib/context/style';
 
 import { registerSW } from './registerSW';
 
+const isServer = !globalThis.document;
 const { BrowserPersistence } = Util;
-const apiBase = new URL('/graphql', location.origin).toString();
+const origin = 'https://magento-venia-concept-e1r4b.local.pwadev:8048';
```

Comment:
This hard-coded value likely only works on your local, this will need some SSR treatment as well since `location.origin` will not be available. Pretty sure you won't have access to `useLocation` outside of the router context either, so some sort of communication from UPWARD would need to inform Apollo of its origin. React Router docs also seem to indicate the need to use the stateless `StaticRouter` instead of `BrowserRouter`, should that be in this scope?

Reply:
True, I should have reworked this. The server will have to provide the origin when it executes the script, so we'll probably want to pull parts of `src/index.js` into a separate function that takes `origin` as an argument.

Reply:
@tjwiebell Okay. I've refactored `Adapter` to follow our talon-component pattern, slimmed down the `index.js` entrypoint, and [set `origin` to `MAGENTO_BACKEND_URL` on the server for now](https://github.com/magento/pwa-studio/pull/3087/commits/7c779ec989f34569faa19150bc32b80f53b8b7e6#diff-dfb401e2af69cac95df4a319d657ddb8771d97bf5a35fd619fc722dec6258e08R12-R15). When we dig into the real SSR work and have an express handler, we can find a way to retrieve `origin` from the HTTP request and get it to our script.